src/modules/credentials/handlers/IssueCredentialHandler.ts

```diff
@@ -1,17 +1,58 @@
+import type { AgentConfig } from '../../../agent/AgentConfig'
 import type { Handler, HandlerInboundMessage } from '../../../agent/Handler'
+import type { CredentialRecord } from '../repository'
 import type { CredentialService } from '../services'
 
+import { createOutboundMessage } from '../../../agent/helpers'
+import { AriesFrameworkError } from '../../../error/AriesFrameworkError'
+import { AutoAcceptCredential } from '../../../types'
+import { CredentialUtils } from '../CredentialUtils'
 import { IssueCredentialMessage } from '../messages'
 
 export class IssueCredentialHandler implements Handler {
   private credentialService: CredentialService
+  private agentConfig: AgentConfig
   public supportedMessages = [IssueCredentialMessage]
 
-  public constructor(credentialService: CredentialService) {
+  public constructor(credentialService: CredentialService, agentConfig: AgentConfig) {
     this.credentialService = credentialService
+    this.agentConfig = agentConfig
   }
 
   public async handle(messageContext: HandlerInboundMessage<IssueCredentialHandler>) {
-    await this.credentialService.processCredential(messageContext)
+    const credentialRecord = await this.credentialService.processCredential(messageContext)
+
+    const autoAccept = CredentialUtils.composeAutoAccept(
+      credentialRecord.autoAcceptCredential,
+      this.agentConfig.autoAcceptCredentials
+    )
+
+    if (autoAccept === AutoAcceptCredential.always) {
+      return await this.nextStep(credentialRecord, messageContext)
+    } else if (autoAccept === AutoAcceptCredential.contentApproved) {
```

Comment:
You haven't removed this from the `processCredential` method:

```
    // Assert the values in the received credential match the values
    // that were negotiated in the credential exchange
    // TODO: Ideally we don't throw here, but instead store that it's not equal.
    // the credential may still have value, and we could just respond with an ack
    // status of fail
    if (credentialRecord.credentialAttributes) {
      CredentialUtils.assertValuesMatch(
        indyCredential.values,
        CredentialUtils.convertAttributesToValues(credentialRecord.credentialAttributes)
      )
    }
```

Reply:
I did not change anything here. I could replace the assertion here with my check, however it states `but instead store that it's not equal`. I am not sure how to handle that specifically.

Reply:
I think we can leave the `but instead store that it's not equal` out and not do the check in here at all. So the record should be updated with the message and the state (As is already happening). Then we have the following possibilities:
- user calls `acceptCredential`. This means they accept the contents of the credential.
- user has `AutoAcceptCredential.always` -> auto accept
- user has `AutoAcceptCredential.contentApproved` -> auto accept (send ack) if credential values match

Reply:
So just remove the if statement and add nothing else if I understand it correctly.

Reply:
Yes